### PR TITLE
Adds typescript-rest framework

### PR DESCRIPTION
**Pull request by jcoleman-techempower**

### CODE REVIEW

File: frameworks/TypeScript/typescript-rest/benchmark_config.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        "plaintext_url": "/plaintext",
+        "update_url": "/updates?queries=",
+        "query_url": "/queries?queries=",
+        "db_url": "/db",
+        "port": 8080,
+        "approach": "Realistic",
+        "classification": "Micro",
+        "framework": "express",
+        "language": "TypeScript",
+        "flavor": "NodeJS",
```
Comment (msmith-techempower) [inline frameworks/TypeScript/typescript-rest/benchmark_config.json]: Shouldn't the `platform` be NodeJS?

File: frameworks/TypeScript/typescript-rest/src/controllers/queries.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    // Use a for-loop here because Array.from is just not
+    // performant at all, but is really nice.
+    // https://jsbench.me/ntjn3s2t0y/1
+    for (let i = 0; i < length; i += 1) {
+      ids.push(randomNumber());
+    }
+
+    const worlds: World[] = await World
+      .query()
+      .findByIds(ids);
```
Comment (msmith-techempower) [inline frameworks/TypeScript/typescript-rest/src/controllers/queries.ts]: I think this is a violation of the rules. You have to incur the cost of querying each id; not as a group.

File: frameworks/TypeScript/typescript-rest/benchmark_config.json (hunk 1 above)
Comment (jcoleman-techempower) [reply frameworks/TypeScript/typescript-rest/benchmark_config.json]: platform?

Comment (msmith-techempower) [reply frameworks/TypeScript/typescript-rest/benchmark_config.json]: Yeah, it's the platform on which the test implementation is built. So, for example, the Gemini test implementation is a Java application written with the Gemini framework which is on the Servlet platform.

File: frameworks/TypeScript/typescript-rest/src/controllers/queries.ts (hunk 2 above)
Comment (jcoleman-techempower) [reply frameworks/TypeScript/typescript-rest/src/controllers/queries.ts]: I'll change this

File: frameworks/TypeScript/typescript-rest/benchmark_config.json (hunk 1 above)
Comment (jcoleman-techempower) [reply frameworks/TypeScript/typescript-rest/benchmark_config.json]: Got it!

**Comment by michaelhixson** [comment]
@msmith-techempower Was the `PR: Awaiting External Approval` label used intentionally?  If so, who are we awaiting approval from?

**Pull request merged**